packages/handler-kit-azure-func/src/function.ts

```diff
@@ -43,8 +43,14 @@ export const azureFunction =
   (
     deps: Omit<R, "logger"> & { inputDecoder: t.Decoder<unknown, I> }
   ): azure.AzureFunction =>
-  (ctx) =>
-    pipe(ctx, azureFunctionTE(h, deps), TE.toUnion)();
+  (ctx) => {
+    const result = pipe(ctx, azureFunctionTE(h, deps), TE.toUnion)();
+    // we have to throws here to ensure that "retry" mechanism of Azure
+    // can be executed
+    if (result instanceof Error) {
```

Comment:
what abouto differentiate between TransientErrors from PermanentErrors?

Reply:
non-permanent errors must be handled at the application level (doing recover via "alt" when necessary)

this code throws an exception only for unhandled errors (therefore automatically "unrecoverable")